conanfile.py

```diff
@@ -1,91 +1,120 @@
 import os
 
-from conans import ConanFile, CMake, RunEnvironment, tools
-from os import path
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMakeToolchain, CMake, CMakeDeps, cmake_layout
+from conan.tools.env import VirtualRunEnv
+from conan.tools.files import copy, load
 
 
 class LibcosimConan(ConanFile):
+    # Basic package info
     name = "libcosim"
+    def set_version(self):
+        self.version = load(self, os.path.join(self.recipe_folder, "version.txt")).strip()
+
+    # Metadata
+    license = "MPL-2.0"
     author = "osp"
-    exports = "version.txt"
-    scm = {
-        "type": "git",
-        "url": "auto",
-        "revision": "auto"
-    }
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "virtualrunenv"
-    requires = (
-        "boost/1.71.0",
-        "fmilibrary/2.3",
-        "ms-gsl/2.1.0",
-        "libzip/1.7.3",
-        "yaml-cpp/0.7.0",
-        "xerces-c/3.2.2",
-        # conflict resolution
-        "openssl/1.1.1k",
-        "xz_utils/5.2.5",
-        "zlib/1.2.12"
-    )
+    url = "https://github.com/open-simulation-platform/libcosim.git"
+    description = "A co-simulation library for C++"
 
+    # Binary configuration
+    package_type = "library"
+    settings = "os", "compiler", "build_type", "arch"
     options = {
         "shared": [True, False],
+        "fPIC": [True, False],
         "proxyfmu": [True, False],
-        "no_fmi_logging": [True, False]}
-    default_options = (
-        "proxyfmu=False",
-        "shared=True",
-        "no_fmi_logging=False"
-    )
-
-    def is_tests_enabled(self):
-        return os.getenv("LIBCOSIM_RUN_TESTS_ON_CONAN_BUILD", "False").lower() in ("true", "1")
+        "no_fmi_logging": [True, False],
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "proxyfmu": False,
+        "no_fmi_logging": False,
+    }
 
-    def set_version(self):
-        self.version = tools.load(path.join(self.recipe_folder, "version.txt")).strip()
+    # Dependencies/requirements
+    def requirements(self):
+        self.tool_requires("cmake/[>=3.19]")
+        self.requires("fmilibrary/[~2.3]")
+        self.requires("libzip/[>=1.7 <1.10]") # 1.10 deprecates some functions we use
+        self.requires("ms-gsl/[>=3 <5]", transitive_headers=True)
+        if self.options.proxyfmu:
+            self.requires("proxyfmu/0.3.2@osp/testing-feature_conan-2")
+            self.requires("boost/[~1.81]", transitive_headers=True) # Required by Thrift
+        else:
+            self.requires("boost/[>=1.71]", transitive_headers=True)
```

Comment:
It seems a consumer of `libcosim` needs to explicitly declare `boost` to use, for example at least `boost::timer` in its code. Otherwise linking fails.

I probably not fully understand the "traits" from the conan doc yet, but may be `transitive_libs` allows the consumers to link boost declared here? So that "downstream" (e.g. libcosim consumers) can also use the same boost version in their code, without accidentally declaring mismatching boost version.

Reply:
The way I see it, consumers who use Boost directly in their own code *should* list it as an explicit dependency and not rely on the indirect dependency from libcosim. However, if they only need Boost to be able to link with libcosim, then it should be handled automatically. I'm actually not sure whether `transitive_libs` is required in that case; I'll try to find out.

Reply:
I *think* the reason I specified `transitive_headers` is because libcosim headers include some Boost headers, so client code must have them on the include path.

Reply:
It turns out you were right about this, @davidhjp01. Since we use Boost in some libcosim headers, and more specifically *Boost libraries that have binary components* (i.e., are not header-only), such as Boost.Log, then we need to have `transitive_libs=True` here. Otherwise, client code won't have access to those binary components for linking. I'll make a new pull request.

Reply:
Should be resolved by #753.